File: frontend/src/api/plan.ts
import { apiClient } from './client'
import { GeneratePlanResponse, ReplanRequest, ReplanResponse } from '../types'

export const planApi = {
	generate: async (centerId: number): Promise<GeneratePlanResponse> => {
		const response = await apiClient.post<GeneratePlanResponse>(
			'/plan/generate',
			{
				center_id: centerId,
			}
		)
		return response.data
	},

	replan: async (data: ReplanRequest): Promise<ReplanResponse> => {
		const response = await apiClient.post<ReplanResponse>(
			'/plan/replan',
			data
		)
		return response.data
	},
}

File: backend/app/api/routes/plan.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.api.deps import get_db
from app.schemas.plan import PlanGenerateRequest, PlanGenerateResponse, ReplanRequest, ReplanResponse
from app.services.replanning import ReplanningService

router = APIRouter(prefix="/plan", tags=["plan"])


@router.post("/generate", response_model=PlanGenerateResponse)
def generate_plan(req: PlanGenerateRequest, db: Session = Depends(get_db)):
    allocations = ReplanningService.generate_plan(db, req.center_id)
    if not allocations:
        return PlanGenerateResponse(allocations=[], message="No unserved sites")
    return PlanGenerateResponse(allocations=allocations)


@router.post("/replan", response_model=ReplanResponse)
def replan_plan(req: ReplanRequest, db: Session = Depends(get_db)):
    result = ReplanningService.replan(db, req.center_id, req.trigger)
    return result
